File: julia/04-control-flow/while_loops.jl
# Classic while loop
countdown = 3
while countdown > 0
    println("T-minus $countdown")
    global countdown -= 1   # 'global' needed at top-level script scope
end
println("Liftoff!")

# break: stop as soon as we find what we want
for n in 1:100
    if n^2 > 50
        println("First square over 50 is $(n^2) (n=$n)")
        break
    end
end

# continue: skip even numbers, print only odds
for n in 1:10
    if iseven(n)
        continue
    end
    print(n, " ")
end
println()

File: python/04-control-flow/while_loops.py
# A simple while loop
count = 3
while count > 0:
    print(f"T-minus {count}")
    count -= 1
print("Liftoff!")

# break and continue in action
print("Even numbers under 10, skipping 4:")
n = 0
while n < 10:
    n += 1
    if n == 4:
        continue        # skip 4
    if n % 2 != 0:
        continue        # skip odd numbers
    if n > 8:
        break           # stop once we pass 8
    print(n)
